src/tiled/abstractobjecttool.cpp

```diff
@@ -302,12 +307,27 @@ void AbstractObjectTool::showContextMenu(MapObjectItem *clickedObjectItem,
     }
 
     if (selectedObjects.size() == 1) {
-        // Saving objects with embedded tilesets is disabled
-        auto cell = selectedObjects.first()->cell();
-        if (cell.isEmpty() || cell.tileset()->isExternal())
-            menu.addAction(tr("Save As Template"), this, SLOT(saveSelectedObject()));
+        MapObject *currentObject = selectedObjects.first();
+        if (!(currentObject->isTemplateBase() || currentObject->isTemplateInstance())) {
```

Comment:
I'm not sure if it makes sense to hide the "Save As Template" action for template instances, given that you could instantiate a template, override some properties, and then want to save the result as another template.

On the other hand, the user may expect one template to derive from another template then, which might be interesting but also a major pain to add support for currently. So maybe it's good to avoid confusion and force a Detach first.

Reply:
Yeah I had similar thoughts, Assuming that we don't want recursive templates, saving would internally detach then save as a template.